core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```
@@ -339,6 +339,177 @@ private void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploy
+            String filePath = configChangeSetRepoPath.getPath();
+            String fileName = filePath.substring(filePath.lastIndexOf('/') + 1);
+            String shellCommand = "./repos/" + fileName + "-master/config-sets/" + configChangeSet.getName();
+            if (isInit) {
+                shellCommand = shellCommand.concat("/apply-config.sh &>/dev/null");
+            } else {
+                shellCommand = shellCommand.concat("/revert-config.sh &>/dev/null");
+            }
+            // Generate array of commands that need to be applied
+            String[] applyShellCommand = { shellCommand };
+            return applyShellCommandOnAgent(testPlan, applyShellCommand);
+        } catch (MalformedURLException e) {
+            logger.warn("Error parsing scenario path " + e);
+            return false;
+        }
+    }
+
+    /**
+     * Initialize agent before running config change set
+     *
+     * @param testPlan  The test plan
+     * @return          True if execution success. Else, false
+     */
+    private boolean initConfigChangeSet(TestPlan testPlan) {
+        try {
+            URL configChangeSetRepoPath = new URL(testPlan.getConfigChangeSetRepository());
+            String filePath = configChangeSetRepoPath.getPath();
+            String fileName = filePath.substring(filePath.lastIndexOf('/') + 1);
+            String[] initShellCommand = {
+                    "mkdir repos",
+                    "cd repos && curl -LJO " +
+                            testPlan.getConfigChangeSetRepository() + "/archive/master.tar.gz   &>/dev/null",
+                    "cd repos && tar xvzf " + fileName + "-master.tar.gz  &>/dev/null",
+                    "chmod -R 755 repos/" + fileName + "-master/config-sets/ &>/dev/null"
+            };
+            return applyShellCommandOnAgent(testPlan, initShellCommand);
+        } catch (MalformedURLException e) {
+            logger.warn("Error parsing scenario path " + e);
```

This will swallow the exception stack trace. `e` should be provided as the second argument. Also, better if we can print which URL failed.